### PR TITLE
bug fixes and new features for cob_light

### PR DESCRIPTION
**Pull request by floweisshardt**
see https://github.com/ipa-fmw/cob_driver/pull/21

new package for light simulation
new light mode 'sweep'
fixed infinit send error

### CODE REVIEW

**Comment by floweisshardt** [comment]
tested on cob4-2: OK

But do not merge (yet):
- still has catkin build erros (dependency problems), see https://github.com/ipa320/cob_driver/issues/212
- Instead of having a new package cob_light_sim, I'd like to merge this into cob_light and offer a sim node inside cob_light

File: cob_light_sim/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Sources
+##############################################################################
+
+file(GLOB_RECURSE QT_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} FOLLOW_SYMLINKS src/*.cpp)
+
+##############################################################################
+# Binaries
+##############################################################################
+
+add_executable(cob_light_sim ${QT_SOURCES} ${QT_RESOURCES_CPP} ${QT_FORMS_HPP} ${QT_MOC_HPP})
```
Comment (fmessmer) [inline cob_light_sim/CMakeLists.txt]: for cob_light_sim executable a `add_dependencies` to cob_light_gencpp is missing
#212

**Comment by fmessmer** [comment]
@ipa-bnm please have a look and adjust accordingly...

**Comment by fmessmer** [comment]
@ipa-bnm 
I'm willing to move the files from `cob_light_sim` to `cob_light` and fix the dependencies as needed.
However, I will need a (preferably minimal) instruction set (i.e. terminal commands) how I can test the `cob_light_sim` functionality.
If you post the commands here, I could then copy them into a README?

**Comment by fmessmer** [comment]
As discussed with @ipa-bnm:
@ipa-fxm will create a new PR only containing the `cob_light` changes of this PR
and @ipa-bnm will further work on `cob_light_sim` and create a dedicated PR for this once work is finalized. This PR then will also integrate `cob_light_sim` within `cob_light`

For the time being keep this PR open until further notice!

**Comment by benmaidel** [comment]
@ipa-fxm @ipa-fmw 
i will reset my indigo_dev branch and create a new clean pull request with the fixes and new features. PR can be closed.

**Comment by floweisshardt** [comment]
superceeded by #220 and #221

**Pull request closed without merging**